llvm/lib/Transforms/Scalar/LoopInterchange.cpp

```diff
@@ -110,6 +110,7 @@ static bool populateDependencyMatrix(CharMatrix &DepMatrix, unsigned Level,
                     << " Loads and Stores to analyze\n");
 
   ValueVector::iterator I, IE, J, JE;
+  SmallDenseSet<size_t> Seen;
```

Comment:
is there any reason we can't use `SmallDenseSet<StringRef>` here? Manually calculate hashes on common types like std::string / StringRef looks a little unusual to me.

Reply:
StringRef doesn't own the data. I construct a std::string in the loop from a vector of characters. My concern is that when that string goes out of scope, the std::string is deallocated, and the StringRef in the set points to invalid memory. I thought just storing an integer would be easiest. And I did a grep for "DenseSmallSet<std::string" and didn't see any precedent for that. Let me know if I got any of this wrong, am of course happy to change this.

Reply:
For a string set that owns the strings you would normally use StringSet.

Reply:
Ah, okay, thanks!